addons/account_facturx/models/account_edi.py

```diff
@@ -39,28 +64,39 @@ def format_monetary(number, currency):
 
         # Create file content.
         template_values = {
-            'record': self,
+            'record': move,
             'format_date': format_date,
             'format_monetary': format_monetary,
         }
-        content = self.env.ref('account_facturx.account_invoice_facturx_export').render(template_values)
-        return b"<?xml version='1.0' encoding='UTF-8'?>" + content
+        xml_content = b"<?xml version='1.0' encoding='UTF-8'?>"
+        xml_content += self.env.ref('account_facturx.account_invoice_facturx_export').render(template_values)
+        xml_name = '%s_facturx.xml' % (move.name.replace('/', '_'))
+        return (xml_name, xml_content)
 
-    def _import_facturx_invoice(self, tree):
-        ''' Extract invoice values from the Factur-x xml tree passed as parameter.
+    @api.model
+    def _decode_xml(self, code, tree, move=None):
+        """ Decodes an xml into an account.move.
+
+        :param code: the code of the type of EDI to decode from.
+        :param tree: the tree to decode.
+        :param move: the account.move to decode to (if None, a new one will be created).
+        :returns: the created account.move if any.
+        """
+        if code != 'facturx':
+            return super()._decode_xml(code, tree, move)
```

Comment:
it should be fine to have that in two methods:

```
if code == 'facturx':
  return self._import_facturx_invoice(filename, xml_tree)
return super()....
```

Reply:
Why not keep it in one method ?

Reply:
It seems cleaner to me because both methods have not the same purpose and should have separated docstrings but it's just my point of view